src/bin/stratisd.rs

```diff
@@ -85,20 +85,45 @@ fn run() -> StratisResult<()> {
 
     let (dbus_conn, mut tree, dbus_context) = libstratis::dbus_api::connect(Rc::clone(&engine))?;
 
-    // Get a list of fds to poll for
-    let mut fds: Vec<_> = dbus_conn
-        .watch_fds()
-        .iter()
-        .map(|w| w.to_pollfd())
-        .collect();
+    let mut fds = Vec::new();
+
+    let mut eventable = engine.borrow_mut().get_eventable()?;
+    let dbus_fd_start_idx = match eventable {
+        Some(ref mut evt) => {
+            fds.push(libc::pollfd {
+                         fd: evt.get_pollable_fd(),
+                         revents: 0,
+                         events: libc::POLLIN,
+                     });
+            1
+        }
+        None => 0,
+    };
+
+    let poll_timeout = if dbus_fd_start_idx == 0 { 10000 } else { -1 };
 
     loop {
-        // Poll them with a 10 s timeout
-        let r = unsafe { libc::poll(fds.as_mut_ptr(), fds.len() as libc::c_ulong, 10000) };
-        assert!(r >= 0);
+        // Handle the Eventable (if there is one) before polling. If an event
```

Comment:
This code still seems very funky, but I feel as if I understand it.

Reply:
it is funky. At some point we may want to look at mio or tokio crates to abstract the messiness of the event loop. I've opened #656 to track.